src/java/org/apache/cassandra/cache/ICache.java

```diff
@@ -23,6 +23,8 @@
  * This is similar to the Map interface, but requires maintaining a given capacity
  * and does not require put or remove to return values, which lets SerializingCache
  * be more efficient by avoiding deserialize except on get.
+ *
+ * This class represents a weighted cache.
```

Comment:
There is also a cache called `ChunkCache' which also has a weighted type, but it doesn't extend the `ICache' interface. So, we either have to fix javadocs for both of these cases or for none of them. The latter case sounds more easy for me as the `CacheSize` already has description.